lib/goth.ex

```diff
@@ -277,6 +269,16 @@ defmodule Goth do
     Process.send_after(self(), :refresh, time_in_seconds * 1000)
   end
 
+  defp get_retry_delay(state) do
+    fun = Map.get(state, :retry_delay, &exp_backoff/1)
```

Comment:
I believe we already set the default in init/1 so doing it here is redundant. Would something like this at the call site work?

`state.retry_delay.(state.retries)`

?

Reply:
tbh, we could remove this private function and call it directly from the `handle_retry/1` with your suggestion. good catch